static/js/src/PurchaseModal/PurchaseModal.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import * as Sentry from "@sentry/react";
+import useStripeCustomerInfo from "./hooks/useStripeCustomerInfo";
+import registerPaymentMethod from "./hooks/registerPaymentMethod";
+import { Formik, FormikHelpers } from "formik";
+import { useQueryClient } from "react-query";
+import { getErrorMessage } from "../advantage/error-handler";
+import { checkoutEvent } from "../advantage/ecom-events";
+import {
+  getUserInfoFromVariables,
+  getInitialFormValues,
+  FormValues,
+} from "./utils/utils";
+import StepOne from "./components/ModalSteps/StepOne";
+import StepTwo from "./components/ModalSteps/StepTwo";
+import { BuyButtonProps } from "./utils/utils";
+type Props = {
+  termsLabel: React.ReactNode;
+  product: any;
```

Comment:
(Optional) There must be a number of things that we know about potential products? Could this be either/or?
Anything more specific than any would be better.

You could also use `unknown` instead, which then you could narrow down with `as` giving much better type checking (and avoiding potential bugs).

https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown
https://2ality.com/2020/06/type-guards-assertion-functions-typescript.html

Reply:
The goal of this modal is to be product agnostic. Not sure how to represent it with TS